pep-0689.rst

```diff
@@ -0,0 +1,231 @@
+PEP: 689
+Title: Semi-stable C API tier
+Author: Petr Viktorin <encukou@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Requires: 523
+Created: 22-Apr-2022
+Python-Version: 3.11
+
+
+Abstract
+========
+
+Some functions and types of the C-API are designated *semi-stable*,
+meaning that they will not change in patch (bugfix/security) releases,
+but may change between minor releases (e.g. between 3.11 and 3.12) without
+deprecation warnings.
+
+The exact set of APIs exposed this way may change in Beta releases.
+
+
+Motivation & Rationale
+======================
+
+The Python C-API is currently divided in three tiers (see
+`devguide <https://devguide.python.org/c-api/>`__ for details):
+
+- Limited API, with high compatibility expectations;
+- Public API, which follows the backwards compatibility policy
+  (PEP 387) which calls for deprecation warnings before changes;
+- Private (internal) API, which can change at any time.
+
+We need a tier between Public and Private to accomodate tools like
+advanced debuggers and JIT compilers, which need access to CPython
+internals but assume the C-API they use does not change in patch releases.
```

Comment:
The current wording is in the rationale section, which is setting out the justification for the proposal. To me, it seems odd to phrase it "We need <feature> because <thing> requires <requirement>, but we don't guarantee requirement.

My suggested rephrasing more explicity spells out that we want to support <thing> properly, there is a reason the current state doesn't work (no guarantees at an API level of stability), and hence <feature> is needed that provides those guarantees.

So this was more a "readability" suggestion, though as author you very much have the right to reject it!

A